app/src/main/java/ch/epfl/sweng/eventmanager/repository/EventRepository.java

```diff
@@ -2,62 +2,67 @@
 
 /**
  * @author Louis Vialar
  */
 @Singleton
 public class EventRepository {
-    // In the future, the events data will come from Firebase
-    // From now, we just get them from a static list
-    private final List<Event> EVENTS = new ArrayList<>(3);
-
-    {
-        EVENTS.add(
-                new Event(1, "Japan Impact", "La plus grande convention sur la culture japonaise " +
-                        "de Suisse Romande !",
-                        new EventOrganizer(1, "PolyJapan", "La commission de l'AGEPoly qui promeut la culture japonaise sur le campus et ses environs", null),
-                        null));
-        EVENTS.add(
-                new Event(2, "Sysmic", "Le festival de musique de l'association des étudiants de Microtechnique",
-                        new EventOrganizer(2, "Sysmic", "L'association des étudiants de Microtechnique", null),
-                        null));
-        EVENTS.add(
-                new Event(3, "Souper de section", "Retrouve tes professeurs et camarades lors du souper de section",
-                        new EventOrganizer(3, "CLIC", "L'association des étudiants en IC", null),
-                        null));
-    }
 
     @Inject
-    public EventRepository() {}
+    public EventRepository() {
+    }
 
     public LiveData<List<Event>> getEvents() {
         final MutableLiveData<List<Event>> data = new MutableLiveData<>();
+        DatabaseReference dbRef = FirebaseDatabase.getInstance().getReference("events");
 
-        data.setValue(Collections.unmodifiableList(EVENTS));
+        dbRef.addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                GenericTypeIndicator<List<Event>> typeToken = new GenericTypeIndicator<List<Event>>() {
+                };
+
+                List<Event> events = dataSnapshot.getValue(typeToken);
+                if (events != null)
+                    events.remove(null); // Somehow the list might contain a null element
+
+                data.postValue(events);
+            }
+
+            @Override
+            public void onCancelled(@NonNull DatabaseError databaseError) {
+
+            }
+        });
 
         return data;
     }
 
     public LiveData<Event> getEvent(int eventId) {
         MutableLiveData<Event> ret = new MutableLiveData<>();
-        for (Event ev : EVENTS) {
-            if (ev.getId() == eventId) {
-                ret.setValue(ev);
-                return ret;
+
+        DatabaseReference dbRef = FirebaseDatabase.getInstance().getReference("events").child(String.valueOf(eventId));
+
+        dbRef.addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                ret.postValue(dataSnapshot.getValue(Event.class));
+            }
+
+            @Override
+            public void onCancelled(@NonNull DatabaseError databaseError) {
```

Comment:
From https://firebase.google.com/docs/database/android/start/?authuser=0, it looks like onCancelled is called when firebase fails to read a value. Can we at least log that something went wrong?